crmsh/xmlutil.py

```diff
@@ -34,13 +35,28 @@ def xmlparse(f):
 
 
 def file2cib_elem(s):
+    cib_tmp_copy = ''
     try:
         f = open(s, 'r')
     except IOError as msg:
-        logger.error(msg)
-        return None
+        logger.info("{} tried to read cib.xml, but : {}".format(userdir.getuser(), msg))
+        cib_tmp_copy = mktemp(suffix=".cib.xml")
+
+    if cib_tmp_copy != '':
+        logger.info("{} gonna try it with sudo".format(userdir.getuser()))
+        # Actually it's not trying to open the file with sudo,
+        # but copying the file with sudo. We do copy,
+        # because xmlparse function requires the function descriptor not the plain text
+        # and this would be so much work to redo it.
+        # It's not too bad, but it's still a workaround and better be refactored, so FIXME!
+        copy_local_file(s, cib_tmp_copy)
```

Comment:
I think creating a readable copy of file, which should have been unreadable, is a vulnerability.